files/default/attachVolume.py

```diff
@@ -57,10 +57,17 @@ def main():
         sys.exit(1)
 
     # Get instance ID
-    instance_id = requests.get("http://169.254.169.254/latest/meta-data/instance-id").text
+    token = requests.put(
+        "http://169.254.169.254/latest/api/token",
+        headers={"X-aws-ec2-metadata-token-ttl-seconds": "300"}
+    )
+    headers = {}
+    if token.status_code == requests.codes.ok:
+        headers["X-aws-ec2-metadata-token"] = token.content
+    instance_id = requests.get("http://169.254.169.254/latest/meta-data/instance-id", headers=headers).text
```

Comment:
At the minimum, this logic representing a precise functionality should be wrapped in a function, e.g. `get_instance_id()`

Reply:
Yes, add a function get_imdsv2_token() in this PR: https://github.com/aws/aws-parallelcluster-cookbook/pull/940 to warp the part of getting token to make it more clear.
I didn't warp get_instance_id() because the token is use both in get instance id and get region, there's only one line of code for get instance id: instance_id = requests.get("http://169.254.169.254/latest/meta-data/instance-id", headers=headers).text